src/test/java/de/tum/cit/aet/artemis/architecture/ResourceArchitectureTest.java

```diff
@@ -34,6 +35,8 @@
 
 class ResourceArchitectureTest extends AbstractArchitectureTest {
 
+    private static final Pattern KEBAB_CASE_PATH_PATTERN = Pattern.compile("^(([a-z0-9\\-.]+|\\{[^}]+})/)*([a-z0-9\\-.]+|\\{[^}]+}|\\*)?$");
```

Comment:
I think I have found a valid regex:

`(\\.?[a-z0-9]+(-[a-z0-9]+)*|\\{[^}]+\\})`:
Matches either a kebab case string with optional period at the beginning. (`\\.?[a-z0-9]+(-[a-z0-9]+)*`)
Or a path parameter (`\\{[^}]+\\}`)

`(/(([a-z0-9]+(-[a-z0-9]+)*|\\{[^}]+\\})))*`:
Same story as above. However everything is preceded by a `/` and the optional period is removed.
This can be repeated as much as needed.

`(\\.json|/\\*)?`:
Optionally allows the path to either end with `.json` or with `/*`

```suggestion
    private static final Pattern KEBAB_CASE_PATH_PATTERN = Pattern.compile("^(\\.?[a-z0-9]+(-[a-z0-9]+)*|\\{[^}]+\\})(/(([a-z0-9]+(-[a-z0-9]+)*|\\{[^}]+\\})))*(\\.json|/\\*)?$");
```

Reply:
Good suggestion, that's probably the cleanest solution :+1: I added it